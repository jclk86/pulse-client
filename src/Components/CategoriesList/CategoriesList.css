.container_categories_list {
  text-align: center;
}
.categories_list {
  list-style: none;
  margin: auto;
  padding-left: 0;
  background-color: blue;
}
.categories_all_link {
  display: block;
  text-decoration: none;
  color: black;
  background-color: white;
  padding-top: 5px;
  padding-bottom: 5px;
  display: block;
}

.categories_all_link:focus {
  margin-left: 3px;
}

.categories_list li {
  width: 100%;
  margin-bottom: 10px;
  font-size: 1.4em;
}
