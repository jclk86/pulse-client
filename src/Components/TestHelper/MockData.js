export const mockUser = {
  username: "Pop123",
  password: "Password123!",
  fullname: "Bob Smith",
  email: "BSmith@gmail.com",
  image_url:
    "https://images.pexels.com/photos/1622419/pexels-photo-1622419.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260",
  profile: "test profile"
};

export const mockArticle = {
  title: "test article",
  content: "test content",
  article_category: "News",
  id: 1,
  date_created: "2019-10-14T21:49:20.570Z",
  author: {
    username: mockUser.username
  }
};

export const mockComment = {
  id: 1,
  content: "test comment content",
  date_created: "2019-10-14T21:49:20.570Z",
  article_id: 1,
  user: {
    id: 1,
    username: mockArticle.author.username
  }
};

export const mockUsers = [
  {
    username: "Pop123",
    password: "Password123!",
    fullname: "Bob Smith",
    email: "BSmith@gmail.com",
    image_url:
      "https://images.pexels.com/photos/1622419/pexels-photo-1622419.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260",
    profile: "test profile"
  },
  {
    username: "testuser2",
    password: "Password123!",
    fullname: "Test User",
    email: "TUser@gmail.com",
    image_url:
      "https://images.pexels.com/photos/1622419/pexels-photo-1622419.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260",
    profile: "test profile"
  }
];

export const mockArticles = [
  {
    title: "test article one",
    content: "test content",
    article_category: "News",
    id: 1,
    date_created: "2019-10-14T21:49:20.570Z",
    author: {
      username: mockUsers[0].username
    }
  },
  {
    title: "test article two",
    content: "test content",
    article_category: "News",
    id: 2,
    date_created: "2019-10-14T21:49:20.570Z",
    author: {
      username: mockUsers[1].username
    }
  }
];

export const categories = [
  { name: "Diary" },
  { name: "Guide" },
  { name: "Advice" },
  { name: "Interview" },
  { name: "Random" },
  { name: "News" }
];
