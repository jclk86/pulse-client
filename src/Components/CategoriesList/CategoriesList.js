import React, { Component } from "react";
import { withRouter, NavLink } from "react-router-dom";
import CategoriesListItem from "../CategoriesListItem/CategoriesListItem";
import PropTypes from "prop-types";
import "./CategoriesList.css";

class CategoriesList extends Component {
  render() {
    const { categories } = this.props;
    return (
      <div className="container_categories_list">
        <div className="container_categories_list_header">
          <h3>Select a Category</h3>
        </div>
        <ul className="categories_list">
          <NavLink to={`/articles`} className="categories_all_link">
            <li>All</li>
          </NavLink>
          <NavLink to={`/Popularity`} className="categories_popularity_link">
            <li>Popularity</li>
          </NavLink>
          {categories.map(category => (
            <CategoriesListItem
              category={category}
              key={category.name}
            ></CategoriesListItem>
          ))}
        </ul>
      </div>
    );
  }
}

CategoriesList.propTypes = {
  categories: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string
    })
  )
};

export default withRouter(CategoriesList);
