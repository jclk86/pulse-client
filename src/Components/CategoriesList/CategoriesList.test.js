import React from "react";
import { BrowserRouter } from "react-router-dom";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { categories } from "../TestHelper/MockData";
import CategoriesList from "./CategoriesList";

describe(`CategoriesList Component`, () => {
  it("renders CategoriesList component without issues", () => {
    const wrapper = shallow(
      <BrowserRouter>
        <CategoriesList categories={categories}></CategoriesList>
      </BrowserRouter>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
