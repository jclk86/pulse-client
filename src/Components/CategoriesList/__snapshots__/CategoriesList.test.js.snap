// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoriesList Component renders CategoriesList component without issues 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <withRouter(CategoriesList)
    categories={
      Array [
        Object {
          "name": "Diary",
        },
        Object {
          "name": "Guide",
        },
        Object {
          "name": "Advice",
        },
        Object {
          "name": "Interview",
        },
        Object {
          "name": "Random",
        },
        Object {
          "name": "News",
        },
      ]
    }
  />
</Router>
`;
